src/app/app-routing.module.ts

```diff
@@ -2,23 +2,32 @@ import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
 const routes: Routes = [
+  {
+    path: 'main',
+    loadChildren: () =>
+      import('./components/main-page/main-page.module').then(
+        (m) => m.MainPageModule
+      ),
+  },
+  {
+    path: 'bookDetails/:id',
```

Comment:
books/:id or products/:id would be more appropriate name for path.

Reply:
changed